pkg/sql/colexec/agg/stddevpop/stddevpop.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2021 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stddevpop
+
+import (
+	"math"
+	"reflect"
+	"strconv"
+	"unsafe"
+
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+)
+
+func String(b []byte) (s string) {
+	pbytes := (*reflect.SliceHeader)(unsafe.Pointer(&b))
+	pstring := (*reflect.StringHeader)(unsafe.Pointer(&s))
+	pstring.Data = pbytes.Data
+	pstring.Len = pbytes.Len
+	return
+}
+
+func (stp *Stddevpop[T1, T2]) TransFloat64(a any) float64 {
+	switch v := a.(type) {
+	case int8:
+		return float64(v)
+	case int16:
+		return float64(v)
+	case int32:
+		return float64(v)
+	case int64:
+		return float64(v)
+	case uint8:
+		return float64(v)
+	case uint16:
+		return float64(v)
+	case uint32:
+		return float64(v)
+	case uint64:
+		return float64(v)
+	case float32:
+		return float64(v)
+	case float64:
+		return v
+	case types.Decimal64:
+		res, _ := strconv.ParseFloat(String(v.Decimal64ToString(stp.inputType.Scale)), 64)
+		return res
+	case types.Decimal128:
+		res, _ := strconv.ParseFloat(String(v.Decimal128ToString(stp.inputType.Scale)), 64)
+		return res
+	default:
+		return 0 //can't reach
+	}
+}
+
+func ReturnType(_ []types.Type) types.Type {
+	return types.New(types.T_int64, 0, 0, 0)
+}
+
+//if is decimal,you need to give pricision
+func New[T1, T2 any](inputType types.Type) *Stddevpop[T1, T2] {
+	return &Stddevpop[T1, T2]{inputType: inputType}
+}
+
+func (sdp *Stddevpop[T1, T2]) Grows(sizes int) {
+	if len(sdp.avgs) == 0 {
+		sdp.avgs = make([]T2, sizes)
+		sdp.sizes = make([]int64, sizes)
+	} else {
+		for i := 0; i < sizes; i++ {
+			var a T2
+			sdp.avgs = append(sdp.avgs, a)
+			sdp.sizes = append(sdp.sizes, 0)
+		}
+	}
+}
+
+func (sdp *Stddevpop[T1, T2]) Eval(vs []T2) []T2 {
+	for i, v := range vs {
+		avg := any(sdp.avgs[i]).(float64) / float64(sdp.sizes[i])
+		variance := any(v).(float64)/float64(sdp.sizes[i]) - math.Pow(avg, 2)
```

Comment:
I am very concerned about all these type case.   Should not be like that -- you have parsed everything to float64, no?

Reply:
Ok, let me optimize above.